### PR TITLE
Dropdown

### PR DESCRIPTION
**Pull request by ravise5**
## Description

Initial PR for the dropdown.

Added Edit Dialog
Added Sightly
Added ReadMe

## Related Issue

https://jira.corp.adobe.com/browse/CQ-4345793

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                    sling:orderBefore="placeholder"
+                                    sling:resourceType="fd/af/authoring/components/granite/form/formsmultifield"
+                                    fieldDescription="Enter Display Values (EnumNames) corresponding to their respective Save values"
+                                    fieldLabel="Display Values - EnumNames">
+                                <field
+                                        jcr:primaryType="nt:unstructured"
+                                        sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                        name="./enumNames"
+                                        required="{Boolean}false"/>
+                            </enumNames>
```
Comment (vdua) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml:86]: Discuss with @im-shiv regarding the correct UI for capturing this

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                    fieldDescription="Must be one of the options."
+                                    fieldLabel="Default Value"
+                                    name="./default"/>
+                            <multiValuePath
+                                    jcr:primaryType="nt:unstructured"
+                                    sling:orderBefore="visible"
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
+                                    fieldLabel="Items Load Path"
+                                    name="./optionsLoadPath"/>
```
Comment (vdua) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml:106]: missing in the docs

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                            sling:resourceType="granite/ui/components/coral/foundation/container">
+                        <items jcr:primaryType="nt:unstructured"/>
+                    </help>
+                    <accessibility
+                            jcr:primaryType="nt:unstructured"
+                            jcr:title="Accessibility"
+                            sling:resourceType="granite/ui/components/coral/foundation/container">
+                        <items jcr:primaryType="nt:unstructured"/>
+                    </accessibility>
+                </items>
```
Comment (vdua) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml:121]: also multiselect attribute must be there in the dialog.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/editor/js/editDialog.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            var richTextEditable = richTextContainer.find(".cq-RichText-editable");
+            var filteredValue = Utils.encodeScriptableTags(textInputValue[0].value);
+            richTextEditable.empty().append(filteredValue);
+            changeFormFields(["./_plainTextValue@Delete", "./_value"], [null, filteredValue]);
+        } else {
+            //Removing html tags from content and setting it to default text field
+            var filteredValue =  $('<div>').html(textInputValue[0].value).text();
+            changeFormFields(["./_value", "./_richTextValue@Delete"], [filteredValue, null]);
+        }
+    }
```
Comment (vdua) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/editor/js/editDialog.js:69]: I am assuming this needs an update

**Review by vdua — CHANGES_REQUESTED** [review]

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml (hunk 2 above)
Comment (ravise5) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml:106]: Okay I will remove this from the dialog.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml (hunk 3 above)
Comment (ravise5) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/_cq_dialog/.content.xml:121]: I'll add this property back

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/editor/js/editDialog.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        TEXTINPUT_AUTOFILL_FIELD_KEYWORD = EDIT_DIALOG + " .cmp-adaptiveform-textinput__autofillfieldkeyword",
+        Utils = window.CQ.FormsCoreComponents.Utils.v1;
+
+
+    /**
+     * Toggles the visibility of the maxLength, minLength, placeholder field based on the checked state of
+     * the allowRichText checkbox
+     * @param {HTMLElement} dialog The dialog on which the operation is to be performed.
+     */
+    function handleRichText(dialog) {
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/editor/js/editDialog.js:36]: this is not required here.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/css.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+#base=css
+textinputview.css
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/css.txt:18]: `dropdownview.css`

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+#base=js
+textinputview.js
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/dropdown/v1/dropdown/clientlibs/site/js.txt:18]: `dropdownview.js`

**Pull request closed without merging**